### PR TITLE
common code for x0 defaults in SPs

### PR DESCRIPTION
**Pull request by whoburg**
This should make things cleaner on the `equality` branch.

ready for review.

### CODE REVIEW

File: gpkit/nomials/nomial_math.py [hunk 1]
```diff
@@ -693,12 +693,20 @@ def as_posyslt1(self):
...
+
+        Order of precedence for x0 default:
+            - substitution value
+            - x0 value
+            - sp_init value
+            - 1.0
+        """
         if x0 is None:
-            x0 = {vk: vk.descr["sp_init"] for vk in varkeys
-                  if "sp_init" in vk.descr}
```
Comment (whoburg) [inline gpkit/nomials/nomial_math.py]: Note that `sp_init` was not getting used for partial subs - I think this was probably a bug

Comment (bqpd) [inline gpkit/nomials/nomial_math.py]: Good point.

**Comment by bqpd** [comment]
Looks great to me, but it's bringing up pylint bugs fixed in another branch. Should we merge?

**Comment by bqpd** [comment]
retest this please

**Pull request merged**